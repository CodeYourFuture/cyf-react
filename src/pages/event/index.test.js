import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

import { EventPage } from './';

const mockEvent = {
  city: 'London',
  description: 'First bit of javascript',
  endDate: '1515851599504',
  intake: 'London Class 3',
  links: [
    {
      'course materials':
        'https://codeyourfuture.github.io/syllabus-master/js-core/week-04/lesson.html',
    },
  ],
  location: 'Ticketmaster',
  mentors: [],
  moduleLeaders: [{ name: 'Kash', avatar: 'something' }],
  startDate: '1515851599504',
  topic: 'Week 4 - Hello Javascript',
};

describe('Event Component', () => {
  it('renders', () => {
    const mockAuth = {
      isAuthenticated: () => true,
      getUserID: () => 'user|12345',
    };
    expect(
      renderer
        .create(
          <MemoryRouter>
            <EventPage {...mockEvent} auth={mockAuth} />
          </MemoryRouter>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
