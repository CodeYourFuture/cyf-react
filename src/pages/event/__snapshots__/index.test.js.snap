// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component renders 1`] = `
<div
  className="css-1ebjmhl"
>
  <div
    className="css-1rheyj1"
  >
    <div
      className="css-1f11qlz"
    >
      <h1 />
      <p
        className="css-a0s8e9"
      >
        <span>
          April 15, 2018
        </span>
        <span>
           - 
        </span>
      </p>
      <div
        className="css-hroent"
      >
        <p>
          <span />
        </p>
      </div>
      <section
        className="css-1thas9u"
      >
        <h5>
          Volunteers: 
          0
        </h5>
        <button
          className="css-1b7napj btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Attend this event
        </button>
      </section>
      <hr
        className="css-1d8iqsy"
      />
      <p
        className="css-ofsyx7"
      />
      <div>
        <a
          className="css-t25ae"
          href="/events"
          onClick={[Function]}
        >
          Back to event listing
        </a>
      </div>
    </div>
  </div>
</div>
`;
